### DISCUSSION TITLE
Publish copilot to OpenVSX

### DISCUSSION THREAD

**Discussion by EsmailELBoBDev2** [community/community#35832] [Copilot Conversations]
It would be nice if you can publish the addon on OpenVSX, so vscodium users can use the addon instead of sideloading it? 

Ref: https://github.com/open-vsx/publish-extensions/issues/595

**Comment by iamcjwilliams** [comment]
@EsmailELBoBDev2 Thank you for your feedback

**Comment by trinitronx** [comment]
Another +1 for this request.

@iamcjwilliams  It seems like it's pretty simple to implement dual publishing via a [GitHub Actions workflow described here][doc]. It probably wouldn't take very long to accomplish so long as the source code is hosted on GitHub.

For better or worse, there currently exists a rift in the ecosystem due to the aforementioned licensing issues (whether they be actual or simply perceived `¯\_(ツ)_/¯` ).  This split between the VisualStudio Marketplace and OpenVSX remains a common source of confusion for new *nix users who simply wish to install and use VSCode like they do on other OS's.  They might wonder why some extensions they are used to using appear to not exist.

So, in order for most Linux developers using stock OSS-packaged versions of VSCode to find extensions such as this one, it's important to also publish to OpenVSX.

I'm sure there are also others who would love to use this extension but currently can't find it due to the split between OpenVSX / VisualStudio Marketplace repos for OSS builds of VSCode.
In the past few years, many OSS Linux packagers of VSCode, VSCodium, (a.k.a. Code OSS) have moved to using [OpenVSX extension Registry][openvsx] and had to [disable VisualStudio Extensions Marketplace by default][1] due to Microsoft's aforementioned licensing restrictions.


[1]: https://www.reddit.com/r/vscode/comments/kb0eb5/extensions_missing/?utm_source=share&utm_medium=ios_app&utm_name=ioscss&utm_content=1&utm_term=1
[openvsx]: https://open-vsx.org/
[doc]: https://github.com/open-vsx/publish-extensions/blob/master/docs/direct_publish_setup.md


Below is some more info from OpenVSX's extension publishing letter template:

# OpenVSX Info / Request Template
Dear extension author, Please publish this extension to the Open VSX marketplace.

## Context

Unfortunately, as Microsoft prohibits usages of the Microsoft marketplace by any other products or redistribution of `.vsix` files from it, in order to use VS Code extensions in non-Microsoft products, we kindly ask that you take ownership of the VS Code extension namespace in [Open VSX](https://open-vsx.org/) and publish this extension on Open VSX.

## What is Open VSX? Why does it exist?

Open VSX is a vendor neutral alternative to the MS marketplace used by most other derivatives of VS Code like [VSCodium](https://vscodium.com/), [Gitpod](https://www.gitpod.io), [OpenVSCode](https://github.com/gitpod-io/openvscode-server), [Theia](https://theia-ide.org/)-based IDEs, and so on.

You can read on about Open VSX at the Eclipse Foundation's [Open VSX FAQ](https://www.eclipse.org/legal/open-vsx-registry-faq/).

## How can you publish to Open VSX?

The docs to publish an extension can be found [here](https://github.com/eclipse/openvsx/wiki/Publishing-Extensions). This process is straightforward and shouldn't take too long. Essentially, you need an authentication token and to execute the `ovsx publish` command to publish your extension. There's also [a doc](https://github.com/open-vsx/publish-extensions/blob/master/docs/direct_publish_setup.md) explaining the whole process with an example GitHub Action workflow.

**Comment by ferocagonzalez** [comment]
Another +1 , we cannot use copilot with anything else but Microsoft marketplace, it would great to have it at openvsx

**Comment by glebbash** [comment]
+1, especially now that [Copilot LSP SDK is available](https://github.blog/changelog/2025-02-10-copilot-language-server-sdk-is-now-available/)

**Comment by ananta-dev** [comment]
The Data Science Team I belong to (at a large insurance company) uses VS Code on Posit Workbench. VS Code on Posit Workbench allows installing extensions from Open VSX marketplace only. 

At my company we have Github Copilot licenses for us to use but I cannot use it because there is no Github Copilot extension on Open VSX.

**Comment by ubuntupunk** [comment]
100% and its 2025

**Comment by prateekkhera-pp** [comment]
+1 to this.

**Comment by chroberino** [comment]
Yet another +1.

**Comment by osideshawn** [comment]
+1 cannot install on AWS Sagemaker Code Editor.

**Comment by pchuong** [comment]
+1000 to this, there are many enterprise customers for our Theia product that want to have github copilot available in the OpenVSX. 

Theia already has support for AI out of the box, can you at least provide an http endpoint for enterprise user to connect to the same backend model that is used by the VSCode github copilot extension?

**Comment by davidrsch** [comment]
This should be done soon taking into consideration that copilot have been announced to be open source in a near future.

**Reply by ananta-dev** [reply]
Hoping so!

**Comment by lucidsushi** [comment]
Yes Please!

**Comment by artbegolli** [comment]
It looks like this was added to OpenVSX https://open-vsx.org/extension/GitHub/copilot-chat up until a few days ago when it was removed - does anyone have any idea about why it was pulled?

**Reply by NawreCode** [reply]
it was still there yesterday :sweat: but looks like the github repo has a new name, it is now [VSCode Copilot Chat](https://github.com/microsoft/vscode-copilot-chat)

**Reply by artbegolli** [reply]
interesting...hoping it gets republished under that new name

**Reply by qcastel** [reply]
I created an issue on their side in case they missed this thread: https://github.com/microsoft/vscode/issues/273508
Feel free to add a 👍 to help bring it to their attention.

**Reply by kelzenberg** [reply]
They stated they didn't publish it on open-vsx and closed your issue @qcastel which makes the whole thing even more mysterious...
